tee-worker/ts-tests/integration-tests/common/utils/assertion.ts

```diff
@@ -244,64 +244,72 @@ export async function assertIdGraphMutationResult(
 
 export async function assertVc(context: IntegrationTestContext, subject: CorePrimitivesIdentity, data: Bytes) {
     const results = context.api.createType('RequestVCResult', data) as unknown as RequestVCResult;
-    const vcHash = results.vc_hash.toString();
-
     // step 1
-    const vcIndex = results.vc_index.toString();
-    const vcRegistry = (await context.api.query.vcManagement.vcRegistry(
-        vcIndex
-    )) as unknown as PalletVcManagementVcContext;
-    const vcStatus = vcRegistry.toHuman()['status'];
-    assert.equal(vcStatus, 'Active', 'Check VcRegistry error:status should be equal to Active');
-
-    // step 2
     // decryptWithAes function added 0x prefix
     const vcPayload = results.vc_payload;
     const decryptVcPayload = decryptWithAes(aesKey, vcPayload, 'utf-8').replace('0x', '');
 
-    const vcPayloadHash = blake2AsHex(Buffer.from(decryptVcPayload));
-    assert.equal(vcPayloadHash, vcHash, 'Check VcPayload error: vcPayloadHash should be equal to vcHash');
-
     /* DID format
     did:litentry:substrate:0x12345...
     did:litentry:evm:0x123456...
     did:litentry:twitter:my_twitter_handle
     */
 
-    // step 3
+    // step 2
+    // check credential subject's DID
     const credentialSubjectId = JSON.parse(decryptVcPayload).credentialSubject.id;
     const expectSubject = Object.entries(JSON.parse(subject.toString()));
 
+    // step 3
     // convert to DID format
     const expectDid = 'did:litentry:' + expectSubject[0][0] + ':' + expectSubject[0][1];
     assert.equal(
         expectDid,
         credentialSubjectId,
-        'Check credentialSubjec error: expectDid should be equal to credentialSubject id'
+        'Check credentialSubject error: expectDid should be equal to credentialSubject id'
     );
 
     // step 4
+    // extrac proof and vc without proof json
     const vcPayloadJson = JSON.parse(decryptVcPayload);
     const { proof, ...vcWithoutProof } = vcPayloadJson;
-    assert.equal(vcIndex, vcPayloadJson.id, 'Check VcIndex error: VcIndex should be equal to vcPayload id');
 
     // step 5
-    const enclaveCount = await context.api.query.teerex.enclaveCount();
+    // prepare teerex enclave registry data for further checks
+    const parachainBlockHash = await context.api.query.system.blockHash(vcPayloadJson.parachainBlockNumber);
+    const apiAtVcIssuedBlock = await context.api.at(parachainBlockHash);
```

Comment:
`parachainBlockNumber` is the block used to calculate the VC, and it won't match the VC's Issuance block. Is it likely that the registered enclave changes between those two instances? better said, should we account for such a case?

Reply:
I think the time of VC's issuance is not important here - it doesn't matter what happens next after VC was generated as long as the VC was generated by legit enclave, but I would love to hear other opinions too.